File: backend/models/category.go
package models

type Category struct {
	ID        int64  `db:"id" json:"id"`
	UserID    int64  `db:"user_id" json:"user_id"`
	Name      string `db:"name" json:"name"`
	Icon      string `db:"icon" json:"icon"`
	Type      int    `db:"type" json:"type"` // 1=支出 2=收入
	SortOrder int    `db:"sort_order" json:"sort_order"`
	CreatedAt string `db:"created_at" json:"created_at"`
}

type CategoryReq struct {
	Name      string `json:"name" binding:"required"`
	Icon      string `json:"icon"`
	Type      int    `json:"type" binding:"required,oneof=1 2"`
	SortOrder int    `json:"sort_order"`
}

File: android/app/src/main/java/com/ledger/origin/model/Category.java
package com.ledger.origin.model;

import java.io.Serializable;

/** 分类 */
public class Category implements Serializable {
    public long id;
    public long user_id;
    public String name;
    public String icon;
    public int type;        // 1=支出 2=收入
    public int sort_order;

    public Category() {}

    public Category(String name, String icon, int type) {
        this.name = name;
        this.icon = icon;
        this.type = type;
    }

    @Override
    public String toString() {
        return name;
    }
}
